src/khoj/database/adapters/__init__.py

```diff
@@ -846,8 +847,59 @@ async def aget_text_to_image_model_config():
         return await TextToImageModelConfig.objects.filter().afirst()
 
 
+class FileObjectAdapters:
+    @staticmethod
+    def overwrite_raw_text(file_object: FileObject, new_raw_text: str):
+        file_object.raw_text = new_raw_text
+
+    @staticmethod
+    def create_file_object(file_name: str, raw_text: str):
+        return FileObject.objects.create(file_name=file_name, raw_text=raw_text)
+
+    @staticmethod
+    def get_file_objects_by_name(file_name: str):
+        return FileObject.objects.filter(file_name=file_name)
+
+    @staticmethod
+    def get_all_file_objects():
+        return FileObject.objects.all()
+
+    @staticmethod
+    def delete_file_object_by_name(file_name: str):
+        return FileObject.objects.filter(file_name=file_name).delete()
+
+    @staticmethod
+    def delete_all_file_objects():
+        return FileObject.objects.all().delete()
+
+    @staticmethod
+    async def async_overwrite_raw_text(file_object: FileObject, new_raw_text: str):
+        await sync_to_async(lambda: setattr(file_object, "raw_text", new_raw_text))()
+        await sync_to_async(file_object.save)()
+
+    @staticmethod
+    async def async_create_file_object(file_name: str, raw_text: str):
+        return await sync_to_async(FileObject.objects.create)(file_name=file_name, raw_text=raw_text)
+
+    @staticmethod
```

Comment:
You can simplify this syntax a little bit. Same with most of the other async methods here, there's usually a corresponding method in the Django ORM that you can use with the `a` prefix that acts as a drop-in replacement. Like `first` -> `afirst`, `create` -> `acreate`, etc.

```suggestion
    @staticmethod
    async def async_create_file_object(file_name: str, raw_text: str):
        return await FileObject.objects.acreate(file=file_name, raw_text=raw_text)

```